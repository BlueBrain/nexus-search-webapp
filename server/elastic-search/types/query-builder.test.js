import makeTypeQuery from "./query-builder";

describe("makeTypeQuery()", () => {
  it(" makes an object with the property aggs", () => {
    expect(makeTypeQuery().aggs).toBeDefined();
  });


  it(" makes an elasticSearch agg object, and the proper with @types, terms, and field", () => {
    expect(makeTypeQuery()).toMatchSnapshot();
  });

  it(" makes an elasticSearch aggs object with a query if a textQuery argument is provided", () => {
    const textQuery = { q: "mySearchString" };
    expect(makeTypeQuery(textQuery).query).toBeDefined();
    expect(makeTypeQuery(textQuery)).toMatchSnapshot();
  });
});
