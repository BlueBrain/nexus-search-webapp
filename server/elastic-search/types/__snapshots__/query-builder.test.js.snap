// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeTypeQuery()  makes an elasticSearch agg object, and the proper with @types, terms, and field 1`] = `
Object {
  "aggs": Object {
    "@types": Object {
      "terms": Object {
        "field": "@type.raw",
      },
    },
  },
  "query": Object {
    "bool": Object {
      "must": Array [],
    },
  },
}
`;

exports[`makeTypeQuery()  makes an elasticSearch aggs object with a query if a textQuery argument is provided 1`] = `
Object {
  "aggs": Object {
    "@types": Object {
      "terms": Object {
        "field": "@type.raw",
      },
    },
  },
  "query": Object {
    "bool": Object {
      "must": Array [
        Object {
          "query_string": Object {
            "query": "(mySearchString* OR mySearchString~)",
          },
        },
      ],
    },
  },
}
`;
